src/api/v2/mod.rs

```diff
@@ -96,43 +96,41 @@ fn ws_route(
 		.and_then(handlers::ws)
 }
 
-async fn publish_message(clients: &WsClients, topic: Topic, message: PublishMessage) {
-	match clients.publish(&topic, message).await {
-		Ok(results) => {
-			let published_messages = results.iter().filter(|&result| result.is_ok()).count();
-			info!(?topic, published_messages, "Message published");
-			for error in results.into_iter().filter_map(Result::err) {
-				warn!(?topic, "Cannot publish message to client: {error}")
-			}
-		},
-		Err(error) => error!(?topic, "Cannot publish message: {error}"),
-	}
-}
-
-pub async fn publish_header_verified(
-	mut header_receiver: broadcast::Receiver<(Header, Instant)>,
+pub async fn publish<T: Clone + TryInto<PublishMessage>>(
+	topic: Topic,
+	mut receiver: broadcast::Receiver<T>,
 	clients: WsClients,
-) {
+) where
+	<T as TryInto<PublishMessage>>::Error: Display,
+{
 	loop {
-		let (header, _) = match header_receiver.recv().await {
+		let message = match receiver.recv().await {
 			Ok(value) => value,
 			Err(error) => {
-				error!("Cannot receive message: {error}");
+				error!(?topic, "Cannot receive message: {error}");
 				return;
 			},
 		};
 
-		let message: HeaderMessage = match header.try_into() {
-			Ok(header) => header,
+		let message: PublishMessage = match message.try_into() {
+			Ok(message) => message,
 			Err(error) => {
-				error!("Cannot conver header: {error}");
+				error!(?topic, "Cannot create message: {error}");
 				continue;
 			},
 		};
 
-		let topic = Topic::HeaderVerified;
-		let message = PublishMessage::HeaderVerified(message);
-		publish_message(&clients, topic, message).await;
+		match clients.publish(&topic, message).await {
+			Ok(results) => {
+				let published = results.iter().filter(|&result| result.is_ok()).count();
+				let failed = results.iter().filter(|&result| result.is_err()).count();
+				info!(?topic, published, failed, "Message published to clients");
+				for error in results.into_iter().filter_map(Result::err) {
```

Comment:
In case of multiple ws clients connecting simultaneously, should we think about aggregating the errors into a single `warn` message?

Reply:
Switched to `debug`.